test/python/dagcircuit/test_dagcircuit.py

```diff
@@ -1657,6 +1668,59 @@ def setUp(self):
         circ1.ccx(self.qr2[0], self.qr2[1], self.qr1[0])
         self.dag1 = circuit_to_dag(circ1)
 
+    def test_unitary_gate_eq(self):
+        """Test equality of equal DAGOpNodes containing unitary gates.
+        See: https://github.com/Qiskit/qiskit-terra/issues/14047
+        """
+        # Create the unitary matrix
+        unitary_matrix = np.array(
+            [
+                [0.65328148 - 0.27059805j, 0.0 + 0.0j, 0.0 + 0.0j, 0.65328148 - 0.27059805j],
+                [0.0 + 0.0j, 0.65328148 - 0.27059805j, -0.65328148 + 0.27059805j, 0.0 + 0.0j],
+                [0.27059805 - 0.65328148j, 0.0 + 0.0j, 0.0 + 0.0j, -0.27059805 + 0.65328148j],
+                [0.0 + 0.0j, 0.27059805 - 0.65328148j, 0.27059805 - 0.65328148j, 0.0 + 0.0j],
+            ]
+        )
+
```

Comment:
I am wondering if its makes sense to replace the specific matrix from the original issue by something simpler to write, e.g. see the other added test ``test_unitary_gate_variants_eq``.

Reply:
IMO it doesn't matter -- this is nice since it recreates the issue and it is clear where it comes from. A simpler matrix is also fine since it'd check the same problem, while no one questioning where these values came from haha